internal/cmdrender/executor.go

```diff
@@ -540,3 +570,148 @@ func pruneResources(hctx *hydrationContext) error {
 	}
 	return nil
 }
+
+type selectionContext struct {
+	rootPackagePath string
+}
+
+// selectInput returns the selected resources based on criteria in selectors
+func selectInput(input []*yaml.RNode, selectors []kptfilev1.Selector, ctx *selectionContext) ([]*yaml.RNode, error) {
+	if len(selectors) == 0 {
+		return input, nil
+	}
+	var filteredInput []*yaml.RNode
+	for _, selector := range selectors {
+		for _, node := range input {
+			match, err := isMatch(node, selector, ctx)
+			if err != nil {
+				return nil, err
+			}
+			if match {
+				filteredInput = append(filteredInput, node)
+			}
+		}
+	}
+	return filteredInput, nil
+}
+
+// isMatch returns true if the resource matches input selection criteria
+func isMatch(node *yaml.RNode, selector kptfilev1.Selector, ctx *selectionContext) (bool, error) {
+	pkgPathMatch, err := packagePathMatch(node, selector, ctx.rootPackagePath)
+	if err != nil {
+		return false, err
+	}
+	// keep expanding with new selectors
+	return nameMatch(node, selector) && namespaceMatch(node, selector) &&
+		kindMatch(node, selector) && apiVersionMatch(node, selector) && pkgPathMatch, nil
+}
+
+// nameMatch returns true if the resource name matches input selection criteria
+func nameMatch(node *yaml.RNode, selector kptfilev1.Selector) bool {
+	return selector.Name == "" || selector.Name == node.GetName()
+}
+
+// namespaceMatch returns true if the resource namespace matches input selection criteria
+func namespaceMatch(node *yaml.RNode, selector kptfilev1.Selector) bool {
+	return selector.Namespace == "" || selector.Namespace == node.GetNamespace()
+}
+
+// kindMatch returns true if the resource kind matches input selection criteria
+func kindMatch(node *yaml.RNode, selector kptfilev1.Selector) bool {
+	return selector.Kind == "" || selector.Kind == node.GetKind()
+}
+
+// apiVersionMatch returns true if the resource apiVersion matches input selection criteria
+func apiVersionMatch(node *yaml.RNode, selector kptfilev1.Selector) bool {
+	return selector.APIVersion == "" || selector.APIVersion == node.GetApiVersion()
+}
+
+// packagePathMatch returns true if the package path of resource matches input selection criteria
+func packagePathMatch(node *yaml.RNode, selector kptfilev1.Selector, rootPackagePath string) (bool, error) {
+	resourcePkgPath, err := pkg.GetPkgPathAnnotation(node)
+	if err != nil {
+		return false, err
+	}
+	return selector.PackagePath == "" || filepath.Join(rootPackagePath, selector.PackagePath) == resourcePkgPath, nil
```

Comment:
hmm...this will cause package portability issue across windows/linux. should we make this path OS-agnostic ? (won't affect linux users and windows support will come for free)

Reply:
This is a wider issue and we do that at different places. This should be addressed as part of windows ready issue. Added a TODO